client/src/pages/profile/index.tsx

```diff
@@ -0,0 +1,170 @@
+import ApplicationLayout from 'layouts/application';
+import Head from 'next/head';
+import { Label, Input } from 'components/forms';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useForm } from 'react-hook-form';
+import { useMutation } from 'react-query';
+import { useCallback } from 'react';
+import { apiService } from 'services/api';
+
+type UserProfilePayload = {
+  fname?: string;
+  lname?: string;
+  email?: string;
+  password?: string;
+};
+
+const schemaValidation = yup.object({
+  fname: yup.string().required('first name is required'),
+  lname: yup.string().required('last name is required'),
+  email: yup.string().email().required(),
+  currentPassword: yup.string().min(8).required('password is required'),
+  newPassword: yup.string().min(8).required('password is required'),
+  passwordConfirmation: yup
+    .string()
+    .required('password confirmation is required')
+    .oneOf([yup.ref('password'), null], 'passwords must match'),
+});
```

Comment:
Every form should have its own schema validation. So, they should be two different schema validation. Also, I recommend creating components to isolate those.